src/snowflake/connector/result_set.py

```diff
@@ -178,19 +178,23 @@ def _fetch_pandas_batches(self, **kwargs) -> Iterator[DataFrame]:
         Snowflake's back-end.
         """
         self._can_create_arrow_iter()
-        return self._create_iter(iter_unit=IterUnit.TABLE_UNIT, structure="pandas")
+        return self._create_iter(
+            iter_unit=IterUnit.TABLE_UNIT, structure="pandas", **kwargs
+        )
 
     def _fetch_pandas_all(self, **kwargs) -> DataFrame:
         """Fetches a single Pandas dataframe."""
-        dataframes = list(self._fetch_pandas_batches())
+        dataframes = list(self._fetch_pandas_batches(**kwargs))
         if dataframes:
+            concat_kwargs = kwargs
+            concat_kwargs.pop("types_mapper", None)
```

Comment:
I think this is going to be error-prone. What if we added `concat_kwargs` differently?
```
    def _fetch_pandas_all(self, concat_kwargs: dict[str, Any], **kwargs) -> DataFrame:
        ...
            **concat_kwargs,
        )
    	# Empty dataframe
    	return self.batches[0].to_pandas(**kwargs)
```
Just be careful, I think `concat_kwargs` were the original kwargs so the API shouldn't change, so maybe add `to_pandas_kwargs` in this manner.

Reply:
Can you take a look the new approach? I tried to avoid changing the customer facing API but also make sure the kwargs will be delivered to the right methods.